File: 프로그래머스/lv2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = []
    L = len(prices)
    for i in range(L-1):
        time = L - i -1
        for j in range(i+1, L):
            if prices[j] < prices[i]:
                time = j - i
                break
        answer.append(time)
    answer.append(0)
    return answer

File: 프로그래머스/lv2/42584. 주식가격/주식가격.java
import java.util.*;

class Solution {
    public int[] solution(int[] prices) {
        int size = prices.length;
        int[] answer = new int[size];
        Stack<Integer> stack = new Stack<>();
        
        for(int i=0; i<size; i++) {
            while(!stack.isEmpty() && prices[stack.peek()] > prices[i]) {
                answer[stack.peek()] = i-stack.pop();
            }
            stack.push(i);
        }
        
        while(!stack.isEmpty()) {
            answer[stack.peek()] = size - 1 - stack.pop();
        }
        
        return answer;
    }
}